src/os_time.h

```diff
@@ -164,6 +163,10 @@ static inline dsec_t  ptime_convert_hr_dsec(hr_time_t hrt){
     return ((dsec_t)((double)hrt * timer_gd.m_1_div_freq ));
 }
 
+static inline dsec_t ptime_convert_ns_dsec(hr_time_t hrt){
+    return (dsec_t)((double)hrt / (1000 * 1000 * 1000));
```

Comment:
We can use here [static_cast<>()](https://en.cppreference.com/w/cpp/language/static_cast), it give benefit of compiler check and has not performance overhead: [link](https://stackoverflow.com/questions/674982/performance-hit-from-c-style-casts).

Reply:
`static_cast<dsec_t>(static_cast<double>(hrt) / (1000 * 1000 * 1000));`